code/modules/recycling/conveyor.dm

```diff
@@ -233,7 +239,7 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	var/static/list/unconveyables = typecacheof(list(/obj/effect, /mob/dead))
 	if(!istype(moving) || is_type_in_typecache(moving, unconveyables) || moving == src)
 		return
-	moving.AddComponent(/datum/component/convey, movedir, 0.2 SECONDS)
+	moving.AddComponent(/datum/component/convey, movedir, speed SECONDS)
```

Comment:
You will need to include the new delay in the logic `conveyable_enter` does to set the any new changes to direction
the way you have things setup currently, starting on a slow belt and entering on a fast one will keep you moving slow, which is wrong

Reply:
If you want to be fully correct, you'd also want some way to "update" an object's speed while it's on the belt, taking into account the time since it last moved.

That's a lot of headache though, so don't feel like it's at all required

Reply:
thanks